ckanext/ontario_theme/datastore.py

```diff
@@ -0,0 +1,111 @@
+from ckan.logic import (
+    tuplize_dict,
+    parse_params,
+)
+from ckan.plugins.toolkit import (
+    ObjectNotFound,
+    NotAuthorized,
+    get_action,
+    _,
+    request,
+    abort,
+    render,
+    c,
+    h,
+)
+import copy
+from .plugin_for_validation import reformat_ui_dict
+
+datastore = Blueprint("datastore", __name__)
+
+
+class DictionaryView(MethodView):
+
+    def _prepare(self, id, resource_id):
+        try:
+            # resource_edit_base template uses these
+            pkg_dict = get_action("package_show")(None, {"id": id})
+            resource = get_action("resource_show")(None, {"id": resource_id})
+            rec = get_action("datastore_search")(
+                None, {"resource_id": resource_id, "limit": 0}
+            )
+            return {
+                "pkg_dict": pkg_dict,
+                "resource": resource,
+                "fields": [f for f in rec["fields"] if not f["id"].startswith("_")],
+            }
+
+        except (ObjectNotFound, NotAuthorized):
+            abort(404, _("Resource not found"))
+
+    def get(self, id, resource_id):
+        """Data dictionary view: show field labels and descriptions"""
+
+        data_dict = self._prepare(id, resource_id)
+
+        # global variables for backward compatibility
+        c.pkg_dict = data_dict["pkg_dict"]
+        c.resource = data_dict["resource"]
+
+        return render("datastore/dictionary.html", data_dict)
+
+    def post(self, id, resource_id):
+        """Data dictionary view: edit field labels and descriptions"""
+        data_dict = self._prepare(id, resource_id)
+        fields = data_dict["fields"]
+        data = dict_fns.unflatten(tuplize_dict(parse_params(request.form)))
+        info = data.get("info")
+        if not isinstance(info, list):
+            info = []
+        info = info[: len(fields)]
+        # create dictionary outside of datastore_create
+        # use a copy of dictionary['fields'] for ui_dict to be reformated
+        dict_fields = {
+            "fields": [
+                {
+                    "id": f["id"],
+                    "type": f["type"],
+                    "info": fi if isinstance(fi, dict) else {},
+                }
+                for f, fi in zip_longest(fields, info)
+            ]
+        }
+        ui_dict_fields = copy.deepcopy(dict_fields)
+
+        """ h.flash_success(
+            _(
+                u'Data Dictionary saved. Any type overrides will '
+                u'take effect when the resource is next uploaded '
+                u'to DataStore'
+            )
+        ) """
+
+        ui_dict = reformat_ui_dict(ui_dict_fields["fields"])
+
+        for row in dict_fields["fields"]:
+            if "info" in row:
+                col_name = row["id"]
+                row["info"]["frictionless_dict"] = [
+                    item for item in ui_dict["fields"] if item.get("name") == col_name
+                ]
+
+        get_action("datastore_create")(
```

Comment:
Add comment above line:
```
# Push the Frictionless data dictionary to database so that 
# ckanext-validation can retrieve it in _run_sync_validation (logic.py)
```